app/scripts/utils/data-utils.ts

```diff
@@ -0,0 +1,24 @@
+import { VedaData, VedaDatum, DatasetData } from '$types/veda';
+
+export function findParentDatasetFromLayer({
```

Comment:
@AliceR  - heads-up that I added this util that is used both for `data-catalog` and `data-catalog-legacy`. (It turns out finding parent dataset on fly is necessary for filtering logic on E&A - this logic is so convoluted at this point but I will not include the change of this logic in this pr.)

Reply:
I will merge this since the pr is pretty big and I am worried about the conflict. lmk if you have any concern about this!